### PR TITLE
[WIP] Feature/image upload

### PR DESCRIPTION
**Pull request by 0xLeif**
Trying to upload an image with Paw doesn't work. I'm interested in if React and upload and retrieve images.

### CODE REVIEW

File: Sources/App/Controllers/ImageController.swift [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import Imperial
+import Authentication
+
+struct ImageController: RouteCollection {
+    let imageFolder = "uploads/"
+    
+    func boot(router: Router) throws {
+        let authSessionRouter = router.grouped(User.authSessionsMiddleware())
+        
+        authSessionRouter.post("uploadImage", use: uploadImageHandler)
```
Comment (ambid17) [inline Sources/App/Controllers/ImageController.swift]: couldn't it still be called image?

File: Sources/App/Controllers/ImageController.swift [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    }
+    
+    func uploadImageHandler(_ req: Request) throws -> Future<ImageUploadResponse> {
+        let user = try req.requireAuthenticated(User.self)
+        return try req.content
+            .decode(ImageUpload.self)
+            .flatMap { imageData in
+                let workPath = try req.make(DirectoryConfig.self).workDir
+                let name = try "\(user.requireID())-\(UUID().uuidString).jpg"
+                let path = workPath + self.imageFolder + name
```
Comment (ambid17) [inline Sources/App/Controllers/ImageController.swift]: couldn't this just be created once on init?

File: Sources/App/Controllers/ImageController.swift [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+                                         attributes: nil)
+                
+                return Future.map(on: req) { ImageUploadResponse(fileName: name) }
+        }
+    }
+    
+    func getImageHandler(_ req: Request) throws -> Future<ImageUpload> {
+        let imageName = try req.parameters.next(ImageUpload.self)
+        
+        let path = try req.make(DirectoryConfig.self).workDir + self.imageFolder + imageName
```
Comment (ambid17) [inline Sources/App/Controllers/ImageController.swift]: looks like this code is reused, could either move it into a utility function, or set up the first part of the path in the init

File: Sources/App/Controllers/UserController.swift [hunk 4]
```diff
@@ -17,13 +17,10 @@ class UserController: RouteCollection {
...
-        authSessionRouter.post("login", use: login)
-        
-        let protectedRouter = authSessionRouter.grouped(RedirectMiddleware<User>(path: "/"))
-        
-        protectedRouter.get("profile", use: profile)
-        protectedRouter.post("updateProfile", use: updateProfile)
         
+        authSessionRouter.post("login", use: login)
+        authSessionRouter.get("profile", use: profile)
+        authSessionRouter.post("updateProfile", use: updateProfile)
```
Comment (ambid17) [inline Sources/App/Controllers/UserController.swift:23]: why isn't this protected?

**Review by ambid17 — APPROVED** [review]
functionality looks good

File: Sources/App/Controllers/ImageController.swift (hunk 1 above)
Comment (0xLeif) [reply Sources/App/Controllers/ImageController.swift]: Yes it could I'll update that

File: Sources/App/Controllers/ImageController.swift (hunk 2 above)
Comment (0xLeif) [reply Sources/App/Controllers/ImageController.swift]: What part? The name var requires user authed. and UUID

File: Sources/App/Controllers/ImageController.swift (hunk 3 above)
Comment (0xLeif) [reply Sources/App/Controllers/ImageController.swift]: I can make it into a util function

File: Sources/App/Controllers/UserController.swift (hunk 4 above)
Comment (0xLeif) [reply Sources/App/Controllers/UserController.swift:23]: API is not handling the UI so the protectedRouter is just `RedirectMiddleware<User>(path: "/")`

**Pull request merged**